File: ExampleLambda.java
// (define evaluate
//     (lambda (f x y)
       // (f x y)))

interface DaveOperation {
    public int f(int a, int b);
}

class Plus implements DaveOperation
    public int f(int a, int b) {
        return a+b;
    }
}

class ExampleLambda {

    public static void evaluate(
            DaveOperation dobj, int x, int y) {
        System.out.println(dobj.f(x,y));
    }

    public static void main(String[] args) {
        evaluate(new DaveOperation(), 3, 5);
    }

}

File: example-lambda.py
def evaluate(f, x, y):
    return f(x,y)

# (define evaluate
#     (lambda (f x y)
#        (f x y)))

def plus(a,b):
    return a+b

print(evaluate(plus, 3, 5))

print(evaluate(lambda x,y: x+y, 3, 5))

# (evaluate
#    (lambda (x y)
#      (+ x y))
#     3
#     5)

people = [("Dave", 3), ("Liz", 8),
          ("Rae", 1), ("Evan", 12)]
people.sort(key=lambda person: person[1])
print(people)

def get_age(person):
    return person[1]

people.sort(key=get_age)
print(people)
